src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -840,8 +840,8 @@ public function findDefinition($id)
      */
     public function createService(Definition $definition, $id, $tryProxy = true)
     {
-        if ('Symfony\Component\DependencyInjection\Definition' !== get_class($definition)) {
-            throw new RuntimeException(sprintf('Constructing service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
+        if ($definition instanceof DefinitionDecorator) {
+            throw new RuntimeException(sprintf('Constructing decorated service "%s" from a %s is not supported at build time.', $id, get_class($definition)));
```

Comment:
this is not a decorated service. It is about using definition inheritance (yes, the name ``DefinitionDecorator`` is quite bad, but the service decoration feature was implemented years later)